ubiquity/pay.ubq.fi
# PR: Unit tests for virtual cards backend

Resolves #271

With unit tests, it also includes some renaming and refactoring to the pages functions code.

### Hot to run unit tests?
```
git clone https://github.com/EresDevOrg/pay.ubq.fi
cd pay.ubq.fi
yarn install
yarn test:unit
```

### QA:
- when all tests pass, workflow should pass: https://github.com/EresDevOrg/pay.ubq.fi/actions/runs/11682621890/job/32530128083
- when a test fails, workflow should fail https://github.com/EresDevOrg/pay.ubq.fi/actions/runs/11682565121/job/32529948500
- test descriptions and results
```

 ✓ tests/unit/get-best-card.test.ts (5)
   ✓ Get best payment card (5)
     ✓ should respond with correct payment card on production
     ✓ should respond with US International Mastercard for Malta as fallback
     ✓ should respond with no payment card for unsupported country
     ✓ should respond with no payment card for low amount permit
     ✓ should respond with correct payment card for sandbox
 ✓ tests/unit/get-order.test.ts (2)
   ✓ Get payment card order (2)
     ✓ should respond with order details
     ✓ should respond with error for invalid order id
 ✓ tests/unit/get-redeem-code.test.ts (3) 1315ms
   ✓ Get payment card redeem code (3) 1303ms
     ✓ should return redeem code 792ms
     ✓ should not return redeem code to user who is not actual buyer
     ✓ should return err when transaction id is empty
 ✓ tests/unit/post-order.test.ts (16) 449ms
   ✓ Post order for a payment card (16) 443ms
     ✓ should post order on production with permit
     ✓ should return err for ordering card that is not best suited
     ✓ should return err for ordering card for unsupported blockchain
     ✓ should return err for ordering card with too low permit amount
     ✓ should return err for ordering card with too high permit amount
     ✓ should return err for ordering card with expired permit
     ✓ should return err order with tx hash that not permit2 interaction
     ✓ should return error with tx hash that is not call to permitTransferFrom
     ✓ should return error with tx hash that transfers wrong token
     ✓ should return error with tx hash that transfers to wrong treasury
     ✓ should post order with uusd
     ✓ should return err with uusd for unsupported chain
     ✓ should return err with uusd for wrong method call
     ✓ should return err with uusd for wrong treasury
     ✓ should post order on sandbox with uusd
     ✓ should post order on sandbox

 Test Files  4 passed (4)
      Tests  26 passed (26)
   Start at  06:59:49
   Duration  5.94s (transform 2.11s, setup 0ms, collect 11.11s, tests 2.16s, environment 1ms, prepare 1.54s)
```
